quipucords/scanner/network/connect.py

```diff
@@ -30,6 +31,77 @@
 logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
 
+# The ConnectTaskRunner creates a new ConnectResultCallback for each
+# credential it tries to connect with, and the ConnectResultCallbacks
+# all forward their information to a single ConnectResultStore.
+class ConnectResultStore(object):
+    """This object knows how to record and retrieve connection results."""
+
+    def __init__(self, scan_task, conn_results):
+        """Get the unique ConnectionResult object for this scan."""
+        self.scan_task = scan_task
+        self.conn_results = conn_results
+
+        source = scan_task.source
+
+        with transaction.atomic():
+            conn_result = conn_results.results.filter(
+                source__id=source.id).first()
+            if conn_result is None:
+                conn_result = ConnectionResult(
+                    scan_task=scan_task, source=source)
+                conn_result.save()
+                conn_results.results.add(conn_result)
+                conn_results.save()
+        self.conn_result = conn_result
+
+        hosts = set((host_range.host_range
+                     for host_range in source.hosts.all()))
+        for sys_result in conn_result.systems.all():
+            hosts.discard(sys_result.name)
```

Comment:
I don't think we want to use the previously run connection results to determine where we pick up. I think we want to start over because we don't know how long the scan has been paused for or how long the server has been down.

Reply:
Done. Thanks for explaining!